### DISCUSSION TITLE
Decrypting and how to not crash browser

### DISCUSSION THREAD

**Discussion by ChaerilM** [livewire/livewire#7960] [1. Help] [answered]
i have files encrypted with sha 256 cbc, the size is up to 50mb for now. when i try to decrypt the file, my browser would crash. how to not crash it.

```

    public function updatedCsv()
    {
        try {
            // read the data
            $data = file_get_contents($this->csv->getRealPath());

            $helper = new \App\Helper\CryptHelper();

            $this->dispatch('refresh-datatable', $helper->decrypt($data))->to('decrypt-list');
        } catch (\Throwable $th) {
            throw $th;
        }
    }
```

**Comment by joshhanley** [comment]
@ChaerilM don't `dispatch()` the decrypted data as you are sending the whole decrypted file back to the browser as a string of data. Store it in a cache or session and then in your other component that is listeneing for the event, get it to pull from cache or session to use it.

Hope this helps!

**Reply by ChaerilM** [reply]
seems that decryption was quick, maybe because it was localhost. but i believe it crash since i tried to print all of those data to component with loop. i hope i could use paginate array / collection. while wondering if wire:stream would work.

i use this for now.
```
    #[On('refresh-datatable')]
    public function refreshDatatable(string $csv)
    {
        $csv = collect(explode(PHP_EOL, $csv))->filter()->toArray();
        $i = 0;
        foreach ($csv as $line) {
            if ($i == 0) {
                $this->headings = str_getcsv($line, '|');
                $this->cols = count($this->headings);
                $i++;

                continue;
            }

            $this->csvData[] = str_getcsv($line, '|');
        }
    }
```

**Reply by ChaerilM** [reply]
i have used session to save decrypted data, and then only show first 20 data for now. the main plan was to see if the data could be decrypted. and if user wants full file, i will just use download decrypted data instead of showing them.

**Reply by ChaerilM** [reply, marked as answer]
in the end i manage to use pagination, took awhile since we don't have paginate from collection. overall speed is somewhat fast. i might just need to add some progress bars to it.

thanks
